### DISCUSSION TITLE
Subscription Payload foreign key values

### DISCUSSION THREAD

**Discussion by lj-n** [supabase/supabase#5958] [Questions] [answered]
With select() it's possible to query foreign tables:
```
const { data, error } = await supabase
  .from('countries')
  .select(`
    name,
    cities (
      name
    )
  `)
```
Is this also possible when receiving a realtime subscription payload?
```
const mySubscription = supabase
  .from('countries')
  .on('*', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()
```
The returned `payload.new` has "only" the foreign keys.
I haven't found anything related in the docs?

**Comment by GaryAustin1** [comment, marked as answer]
Realtime will only return the row of the table that changed.  If you need other foreign tables you will need to do a select with foreign tables included based on the id in the payload, or fetch the foreign table data from the key.  This would be done in the payload handler.

**Reply by lj-n** [reply]
Alright, thanks for the quick reply!
